dht.go

```diff
@@ -312,21 +312,20 @@ func (dht *IpfsDHT) betterPeersToQuery(pmes *pb.Message, p peer.ID, count int) [
 
 	// no node? nil
 	if closer == nil {
+		log.Warning("no closer peers to send:", p)
 		return nil
 	}
 
-	// == to self? thats bad
-	for _, p := range closer {
-		if p == dht.self {
-			log.Debug("attempted to return self! this shouldn't happen...")
-			return nil
-		}
-	}
-
 	var filtered []peer.ID
 	for _, clp := range closer {
+
+		// == to self? thats bad
+		if clp == dht.self {
+			log.Warning("attempted to return self! this shouldn't happen...")
```

Comment:
Wouldn't that happen if we are in the closest 20 (?) peers, or is it filtered out too?